bundles/de.uka.ipd.sdq.simucomframework/src/de/uka/ipd/sdq/simucomframework/resources/SimulatedResourceContainer.java

```diff
@@ -140,6 +140,41 @@ private IPassiveResource getSimplePassiveResource(final PassiveResource resource
             final AssemblyContext assemblyContext, final SimuComModel simuComModel, final long capacity) {
         return new SimSimpleFairPassiveResource(resource, assemblyContext, simuComModel, capacity);
     }
+    
+    /**
+     * Demand processing of a resource demand by a given type of active
+     * resources In future versions this has to control schedulers of resource
+     * types which exist in multiple instances, use parent container when resource not found
+     *
+     * @param requestingProcess
+     *            The thread requesting the processing of a resource demand
+     * @param resourceServiceID
+     *            the id of the resource service to be called.
+     * @param typeID
+     *            ID of the resource type to which the demand is directed. Same
+     *            as the PCM resource type IDs
+     * @param demand
+     *            The demand in units processable by the resource. The resource
+     *            is responsible itself for converting this demand into time
+     *            spans
+     */
+    public void loadActiveResource(final SimuComSimProcess requestingProcess, final int resourceServiceID,
+            final String typeID, final double demand) {
+    	  try {
+              super.loadActiveResource(requestingProcess, resourceServiceID, typeID, demand);
+          } catch (final ResourceContainerIsMissingRequiredResourceType e) {
+              if (this.parentResourceContainer == null) {
+                  if (LOGGER.isEnabledFor(Level.ERROR)) {
+                      LOGGER.error("Resource container is missing a resource which was attempted to be loaded"
+                              + " by a component and has no parent Resource Container to look in. ID of resource type was: "
+                              + typeID);
+                  }
+                  throw new ResourceContainerIsMissingRequiredResourceType(typeID);
+              } else {
+                  this.parentResourceContainer.loadActiveResource(requestingProcess, resourceServiceID, typeID, demand);

```

Comment:
The only thing that could be confusing is, that missing resource exceptions will further more only be thrown for the outer most resource container. Making it hard to trace back to the inner one, where the exception originally arose. We might be ok with that (seeing that its a special case anyhow). On the other hand if we imagine very many vms / pods inside a big host with particular local resources, the resulting exception would have very little information. 

We could simply catch this exception from the super implementation and rethrow the original one instead.